.github/workflows/docker_image.yml

```
@@ -0,0 +1,87 @@
+---
+name: Docker Image
+
+# https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
+# https://docs.docker.com/build/ci/github-actions/multi-platform
+# https://github.com/opencontainers/image-spec/blob/main/annotations.md
+
+on:
+  push:
+    branches: [ "master" ]
+    # Publish semver tags as releases.
+    tags: [ "*.*.*" ]
+  pull_request:
+    # Build docker image on pull requests to master. (but do not publish)
+    branches: [ "master" ]
+    paths:
+      - 'lib/**.[ch]'
+      - 'lib/**.cpp'
+      - 'src/**.[ch]'
+      - 'src/**.cpp'
+      - '**/CMakeLists.txt'
+      - 'cmake/Modules/**'
+      - 'util/ci/**'
+      - 'misc/irrlichtmt_tag.txt'
+      - 'Dockerfile'
+      - '.dockerignore'
+      - '.github/workflows/docker_image.yml'
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v3.0.0
+
+      # Login against the Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3.0.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Extract metadata (tags, labels) for Docker
+      # https://github.com/docker/metadata-action
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v5.5.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          labels: |
+            org.opencontainers.image.title=Minetest
+
+      # Build and push Docker image
+      # https://github.com/docker/build-push-action
+      # No arm support for now. Require cross-compilation support in Dockerfile to not use QEMU.
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5.1.0
+        with:
+          context: .
+          platforms: linux/amd64
+          push: ${{ github.event_name != 'pull_request' }}
+          load: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Test Docker Image
+        run: docker run --rm $(echo ${{ steps.meta.outputs.tags }} | cut -d',' -f1) minetestserver --version
+        shell: bash
```

cleaner IMO
(assuming this works, I didn't test it of course)
```suggestion
        run: docker run --rm $(cut -d, -f1 <<<"$IMAGE_TAGS") minetestserver --version
        shell: bash
        env:
          - IMAGE_TAGS: "${{ steps.meta.outputs.tags }}"
```